cosmos/dbt/graph.py

```diff
@@ -143,33 +144,39 @@ def load_via_dbt_ls(self) -> None:
         if self.select:
             command.extend(["--select", *self.select])
 
-        with self.profile_config.ensure_profile() as (profile_path, env_vars):
-            command.extend(
-                [
-                    "--profiles-dir",
-                    str(profile_path.parent),
-                    "--profile",
-                    self.profile_config.profile_name,
-                    "--target",
-                    self.profile_config.target_name,
-                ]
-            )
+        with tempfile.TemporaryDirectory() as tmpdir:
+            # when we drop support to Python 3.7, we can use:
+            # shutil.copytree(self.project.dir, tmpdir, dirs_exist_ok=True)
+            cwd = Path(tmpdir) / self.project.name
+            shutil.copytree(self.project.dir, cwd)
+
+            with self.profile_config.ensure_profile() as (profile_path, env_vars):
+                command.extend(
+                    [
+                        "--profiles-dir",
+                        str(profile_path.parent),
+                        "--profile",
+                        self.profile_config.profile_name,
+                        "--target",
+                        self.profile_config.target_name,
+                    ]
+                )
 
-            env = os.environ.copy()
-            env.update(env_vars)
-
-            logger.info("Running command: `%s`", " ".join(command))
-            logger.info("Environment variable keys: %s", env.keys())
-            process = Popen(
-                command,
-                stdout=PIPE,
-                stderr=PIPE,
-                cwd=self.project.dir,
-                universal_newlines=True,
-                env=env,
-            )
+                env = os.environ.copy()
+                env.update(env_vars)
+
+                logger.info("Running command: `%s`", " ".join(command))
+                logger.info("Environment variable keys: %s", env.keys())
+                process = Popen(
+                    command,
+                    stdout=PIPE,
+                    stderr=PIPE,
+                    cwd=cwd,
+                    universal_newlines=True,
+                    env=env,
+                )
 
-            stdout, stderr = process.communicate()
+                stdout, stderr = process.communicate()
```

Comment:
if logs do get written, how would a user access them?

Reply:
If users care about logs, they should define `DBT_LOG_PATH`

Reply:
I disagree - logs are written when the parsing fails. If I'm setting up Cosmos, and the parsing fails, I won't have set up my own `DBT_LOG_PATH` in advance. Instead, I would expect to see the logs of _what_ exactly failed so I could troubleshoot.

IMO the desired behavior is: we set the log and target path for the user; if, at the end of execution, there's something in the log folder (i.e. the command failed and dbt wrote logs), we should print those logs so a user can troubleshoot on their own. If we don't print the logs, a user won't be able to access them because they get cleaned up via the temp dir.

Reply:
Just implemented this logic, @jlaneve